simulation/simulation.py

```diff
@@ -0,0 +1,295 @@
+import asyncio
+import websockets
+import json
+import random
+import time
+import tkinter as tk
+import math
+from tornado import httpclient, gen
+
+import parking.shared.ws_models as wsmodels
+import parking.shared.rest_models as restmodels
+from parking.shared.util import serialize_model
+
+resturl = 'http://127.0.0.1:5000/spaces'
+
+class SimManager:
+    def __init__(self, no_spaces, min_spaces_per_lot, max_spaces_per_lot, no_cars,
+                 x, y, parking_lot_seed, car_seed, max_time):
+        self.random_lot = random.Random(parking_lot_seed)
+        self.random_car = random.Random(car_seed)
+        self.x = x
+        self.y = y
+        self.no_cars = no_cars
+        self.car_tasks = []
+        self.space_tasks = []
+        self.max_time = max_time
+        self.cars = []
+        self.lots = []
+
+        count = 0
+        name = 0
+        while count < no_spaces:
```

Comment:
This appears to have no way of _stopping_ a simulation. 
It would be good to move the actual loop into a coroutine with a yield point on each loop.

For testing it would be nice to be able to say something like:

    sim = SimManager(*state)
    asyncio.ensure_future(sim.run())
    await asyncio.sleep(3)
    await sim.stop()
    assert some_test_on_the_sim_state(sim)
   
Then we could run simulations, stop them and extract data from it afterwards.